sdk/communication/communication-administration/src/phoneNumber/lroModels.ts

```diff
@@ -0,0 +1,248 @@
+ */
+export interface UpdatePollerOptions<T> {
+  abortSignal?: AbortSignalLike;
+  fireProgress?: (state: T) => void;
+}
+
+/**
+ * @ignore
+ * Represents options for creating an instance of ReleasePhoneNumbersPoller
+ */
+export interface ReleasePhoneNumbersPollerOptions extends PhoneNumberPollerOptions {
+  /**
+   * The list of phone numbers to be released.
+   */
+  phoneNumbers: string[];
+
+  /**
+   * Represents the poller client used internally.
+   */
+  client: PhoneNumberPollerClient;
+
+  /**
+   * Additional request options.
+   */
+  options?: ReleasePhoneNumbersOptions;
+}
+
+/**
+ * @ignore
+ * Represents options for creating an instance of ReservePhoneNumbersPoller
+ */
+export interface ReservePhoneNumbersPollerOptions extends PhoneNumberPollerOptions {
+  /**
+   * Request to create reservation.
+   */
+  reservationRequest: CreateReservationRequest;
+
+  /**
+   * Represents the poller client used internally.
+   */
+  client: PhoneNumberPollerClient;
+
+  /**
+   * The id returned from the create reservation request.
+   */
+  reservationId?: string;
+
+  /**
+   * Options for creating a phone number reservation.
+   */
+  options?: CreateReservationOptions;
+}
+
+/**
+ * @ignore
+ * Represents options for creating an instance of PurchaseReservationPoller
+ */
+export interface PurchaseReservationPollerOptions extends PhoneNumberPollerOptions {
+  /**
+   * The id returned from the create reservation request.
+   */
+  reservationId: string;
+
+  /**
+   * Represents the poller client used internally.
+   */
+  client: PhoneNumberPollerClient;
+
+  /**
+   * Options for creating a phone number reservation.
+   */
+  options?: CreateReservationOptions;
+}
+
+/**
+ * @ignore
+ * Represents the poller client used internally
+ */
+export interface PhoneNumberPollerClient {
+  releasePhoneNumbers: (
+    phoneNumbers: string[],
+    options?: StartReleasePhoneNumbersOptions
+  ) => Promise<ReleasePhoneNumbersResponse>;
+  getRelease: (releaseId: string, options?: GetReleaseOptions) => Promise<GetReleaseResponse>;
+  createReservation(
+    reservationRequest: CreateReservationRequest,
+    options: CreateReservationOptions
+  ): Promise<CreateReservationResponse>;
+  getReservation(reservationId: string, options: OperationOptions): Promise<GetReservationResponse>;
+  cancelReservation(
+    reservationId: string,
+    options: CancelReservationOptions
+  ): Promise<VoidResponse>;
+  purchaseReservation(
+    reservationId: string,
+    options: PurchaseReservationOptions
+  ): Promise<VoidResponse>;
+}
+
+/**
+ * @ignore
+ * Represents the operation state of the release phone numbers poller
+ */
+export interface ReleasePhoneNumbersPollOperationState
+  extends PollOperationState<PhoneNumberRelease> {
+  /**
+   * The list of phone numbers to be released.
+   */
+  phoneNumbers: string[];
+
+  /**
+   * Represents the poller client used internally.
+   */
+  client: PhoneNumberPollerClient;
```

Comment:
Hi! I know we've been using this pattern in other clients, but the client's don't need to live as part of the operation state. The client can be passed only through parameters. I'll make a PR today to showcase this. Please ping me if you don't hear of me today.

Reply:
I've added you as a reviewer to this PR: https://github.com/Azure/azure-sdk-for-js/pull/11717 it's a cleanup of the Key Vault Keys LROs, based on a recent brainstorming session I had with @willmtemple ! Reviews appreciated 🙏